kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/validation/impl/IncomeTypeTotalsValidation.java

```diff
@@ -0,0 +1,110 @@
+package edu.arizona.kfs.module.purap.document.validation.impl;
+
+import org.kuali.kfs.module.purap.document.PurchasingAccountsPayableDocument;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.document.validation.GenericValidation;
+import org.kuali.kfs.sys.document.validation.event.AttributedDocumentEvent;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.arizona.kfs.module.purap.businessobject.CreditMemoIncomeType;
+import edu.arizona.kfs.module.purap.businessobject.PaymentRequestIncomeType;
+import edu.arizona.kfs.module.purap.document.PaymentRequestDocument;
+import edu.arizona.kfs.module.purap.document.VendorCreditMemoDocument;
+
+/**
+ * Validate Income Types (1099 Classification tab) total against document total.
+ */
+public class IncomeTypeTotalsValidation extends GenericValidation {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(IncomeTypeTotalsValidation.class);
+
+    public static final String ERROR_INVALID_INCOME_TYPES_TOTAL_AMOUNT = "errors.document.ap.incomeTypesTotalAmount.invalid";
+    
+    public boolean validate(AttributedDocumentEvent event) {
+        boolean isValid = true;
+        GlobalVariables.getMessageMap().clearErrorPath();
+        GlobalVariables.getMessageMap().addToErrorPath(KFSPropertyConstants.DOCUMENT);
+        
+        PurchasingAccountsPayableDocument purchasingDocument = (PurchasingAccountsPayableDocument) event.getDocument();
+        if (purchasingDocument instanceof PaymentRequestDocument) {
+            isValid = validatePaymentRequest((PaymentRequestDocument) purchasingDocument);
+        }
+        else if (purchasingDocument instanceof VendorCreditMemoDocument) {
+            isValid = validateCreditMemo((VendorCreditMemoDocument) purchasingDocument);
+        }
+        
+        GlobalVariables.getMessageMap().removeFromErrorPath(KFSPropertyConstants.DOCUMENT); 
+        
+        //if validation in batch, just log error message, don't fail
+        if (!isValid && GlobalVariables.getUserSession().getPrincipalName().equals(KFSConstants.SYSTEM_USER)) {
+            LOG.info("validateIncomeTypeTotals() Document " + purchasingDocument.getPurapDocumentIdentifier() + ", " + ERROR_INVALID_INCOME_TYPES_TOTAL_AMOUNT);
```

Comment:
Comments mention "log error message", but logging at the Info level... Should this log at the Error level, or is logging at Info with the defined error message sufficient?